src/clr-angular/data/datagrid/built-in/filters/datagrid-numeric-filter-impl.spec.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { ClrDatagridNumericFilterInterface } from '../../interfaces/numeric-filter.interface';
+
+import { DatagridNumericFilterImpl } from './datagrid-numeric-filter-impl';
+import { DatagridPropertyNumericFilter } from './datagrid-property-numeric-filter';
+import { ClrDatagridFilterInterface } from '../../interfaces/filter.interface';
+import { Observable } from 'rxjs';
+
+export default function(): void {
+  describe('DatagridNumericFilterImpl', function() {
+    let fullFilter: DatagridNumericFilterImpl<number>;
+
+    beforeEach(function() {
+      const numericFilter = new TestFilter();
+      fullFilter = new DatagridNumericFilterImpl(numericFilter);
+    });
+
+    it('becomes active when either high or low limits are set', function() {
+      expect(fullFilter.isActive()).toBe(false);
+      fullFilter.high = 10;
+      expect(fullFilter.isActive()).toBe(true);
+      fullFilter.high = null;
+      expect(fullFilter.isActive()).toBe(false);
+      fullFilter.low = 1;
+      expect(fullFilter.isActive()).toBe(true);
+      fullFilter.low = null;
+      expect(fullFilter.isActive()).toBe(false);
+    });
+
+    it('filters numbers that are above the high limit', function() {
+      expect(fullFilter.accepts(11)).toBe(true);
+      fullFilter.high = 10;
+      expect(fullFilter.accepts(11)).toBe(false);
+      expect(fullFilter.accepts(9)).toBe(true);
+      fullFilter.high = null;
+      expect(fullFilter.accepts(11)).toBe(true);
+    });
+
+    it('filters numbers that are below the low limit', function() {
+      expect(fullFilter.accepts(0)).toBe(true);
+      fullFilter.low = 1;
+      expect(fullFilter.accepts(0)).toBe(false);
+      expect(fullFilter.accepts(2)).toBe(true);
+      fullFilter.low = null;
+      expect(fullFilter.accepts(0)).toBe(true);
+    });
+
+    it('only allows values within the range when both filters are set', function() {
+      fullFilter.low = 1;
+      fullFilter.high = 10;
+      expect(fullFilter.accepts(0)).toBe(false);
+      expect(fullFilter.accepts(5)).toBe(true);
+      expect(fullFilter.accepts(12)).toBe(false);
+    });
+
+    // TODO: why does this throw an error on build?
+    // it('exposes state', function() {
```

Comment:
What was the error in the build?

Reply:
There was actually nothing wrong with the test, I missed something in the `state` function.  The test now works.